### DISCUSSION TITLE
Is GLMsingle really a wholesale replacement of its predecessor, GLMdenoise?

### DISCUSSION THREAD

**Discussion by Aakashagr** [cvnlab/GLMsingle#122] [Show and tell]
Having worked with both GLMsingle and GLMdenoise, I believe that GLMdenoise is still relevant for experiment designs where the stimulus is repeated multiple times within a given run. My goal here is to compare the reliability of the beta values after denoising using either of the two approaches. The analysis is performed on one good subject with very small head movements. The details of my experiment design. 

**Figure 1**
![image](https://github.com/cvnlab/GLMsingle/assets/12946897/32d3279f-b060-45d4-8a58-8f957432e833)

When I use GLMsingle to estimate the beta values. It changes the design matrix (as shown above) to designSINGLE where one column corresponds to each stimulus presentation. As expected in GLMsingle, the beta values estimated from "FITHRF GLMdenoise RR" gives the highest reliability score - correlation between the odd and even indices
 
**Figure 2**
![image](https://github.com/cvnlab/GLMsingle/assets/12946897/ea5fed53-0356-42a7-8731-05c5064f1ffc)

However, when I use the original design matrix (Figure 1) and estimate the beta values using SPM, the reliability scores are much higher than GLMsingle even without any denoising. and using GLMdenoise package further improves the reliability scores

** Figure 3 **
![image](https://github.com/cvnlab/GLMsingle/assets/12946897/7af4ddbd-7bfe-4d4e-aed6-0b476d2636e0)

Although, this needs to further tested with a large group of subjects and task conditions, it is well known that the estimates of the unknown weights in regression are stable with more observations per condition. Thus GLMsingle does not replace GLMdenoise in my opinion. 

Looking forward to hear the thoughts of other researchers

Regards,
Aakash

p.s.  The reliability of the beta estimates can be further improved by using [NORDIC](https://github.com/SteenMoeller/NORDIC_Raw/tree/main) in addtion to GLMdenoise.

**Comment by kendrickkay** [comment]
Thank you for the information-rich message.  In order for me to fully understand, a few questions:

So, in your experiment, do all repeats for a given condition occur fully within a single run?

Do you have the GLMsingle figure outputs (and/or the GLMdenoise figure outputs) handy that we can stare at to gather further information and diagnostics?

Can you explain exactly how you quantify reliability? Do you split odd/even trials within each condition?  For the approach in which you fit a design matrix with all trials for each condition in a single column, you get only one beta per condition, so how do you split that?

Regarding NORDIC, imposing low rank on a given dataset makes other assumptions (e.g. makes structures more similar locally within a neighborhood of voxels).  These types of assumptions target types of noise that are distinct from those targeted by GLMdenoise/GLMsingle.  There are some risks of such approaches; you want want to take a look at https://doi.org/10.1371/journal.pone.0270895 and/or https://www.sciencedirect.com/science/article/pii/S1053811923002641

**Reply by Aakashagr** [reply]
Hi Kendrick, 
1) No. As described in Figure 1 (in original message), there are 4 runs, and each condition (n= 36) is repeated 5 times within each run. This leads to a total of 20 repeats/condition during the entire experiment session. All runs are identical except for stimulus order. 

2) Yes, please find the figures [here](https://drive.google.com/drive/folders/1dPr_wEdgGO8FpjqtpK4uoRQsYn48Vi6R?usp=drive_link). There are 3 folders: GLMsingle, GLMsingle implemented after NORDIC, and GLMdenoise implemented after NORDIC. Unfortunately, I forgot to rename the GLMdenoise without NORDIC analysis, and the figures were overwritten. But the trends are qualitatively similar. 
I can regenerate them as well if required.

3) First, I identified ~5000 voxels in visual areas that respond to my chosen stimuli. The output of GLMsingle mat files are rearranged as [nvoxels (5000) x nconditions (36) x nrepeats (20)]. For each voxel, I estimate the reliability using split-half measure i.e.,  correlation  between the mean activation pattern across 36 conditions averaged across odd indices (n = 10) and even indices (n = 10). 
With GLMdenoise, I first reconstructed back the denoised nii file, and then used SPM 12 to estimate beta values. The design matrix that is provided to SPM is show in Figure 1 (in original message). Note that only the repeats within a given run are merged but not the runs themselves. Thus, for each condition, I get 4 beta values across 4 runs. To estimate the reliability, I again estimate split-half correlations i.e., average the activations across odd indices (n = 2) and even indices (n= 2) and then estimate correlation coeffiicient

4) Yes, I agree with you and thanks for sharing those articles. I will go through them in detail soon.

Please let me know if you need any further clarifications

Aakash

**Reply by kendrickkay** [reply]
Some thoughts and comments on figures are in this video: https://www.dropbox.com/s/ab3u0vw1lt0bq9q/20231221%20glmsingle%20glmdenoise%20Aakashagr%20data.mp4?dl=0

**Reply by Aakashagr** [reply]
Hi Kendrick, 

Thanks for taking out time and going through the figures in detail. 

Sorry, the design matrix was not labeled clearly and it was misleading. It actually contains all 4 runs as described in the figure below
![image](https://github.com/cvnlab/GLMsingle/assets/12946897/80be0013-4f3f-4b8b-a9eb-6b9146981f09)

The signal bias in the lateral regions is because  "Dielectric pads were placed around the ear to reduce the signal dropout around the anterior Ventral Occipital Temporal cortex."  You can find details of the experiment and fMRI data in the following[ biorxiv paper](https://www.biorxiv.org/content/10.1101/2023.11.06.565812v1.full). Search for the term "Data acquisition"

I agree that the reliability of the data is quite low, perhaps due to the nature of stimuli. Unlike NSD, which has a very diverse stimulus set, my experiment has only 3 letter strings which would evoke very similar activity values within a given voxel. However, the reliability at the population level, i.e. RSA type analysis, is quite robust, especially in early visual areas due to the change in retinotopic location of the stimuli (Figure S12 of [Supplementary material](https://www.biorxiv.org/content/10.1101/2023.11.06.565812v1.supplementary-material).

In my main analysis, I merged all 4 runs and estimated a single beta value per condition. I think the beta values from SPM and GLMdenoise are similar and would not change the conclusions. I just got stuck to my old habits of using GLMdenoise and SPM after a quick check that GLMsingle is not well suited for my experiment design. 

Only recently, I got a chance to do a more systematic comparison of these different denoising techniques (including NORDIC). I also made a similar observation of GLMdenoise being better in another 3T study with a very similar design but with audio stimulus (i.e., different phonemes were presented briefly).

Hence, I wonder if GLMsingle is well suited for only certain experiment designs. I am happy to share the raw data and discuss it over call as well to explore this in more detail.

**Reply by kendrickkay** [reply]
Hello - Not sure exactly what you have in mind or are thinking, but I am happy to discuss if you like.  You can use this site: https://kendrickkay.youcanbook.me
